src/translators/crate.py

```diff
@@ -63,7 +63,10 @@ def setup(self):
         url = "{}:{}".format(self.host, self.port)
         self.conn = client.connect([url], error_trace=True)
         self.cursor = self.conn.cursor()
-
+        # IMPORTANT this reduce queries to crate, but it also means
+        # that you need to redeploy ql in case of major crate update
```

Comment:
Are you saying this because of global state? There's no global state here, you'll get a fresh translator instance each time the notify endpoint gets hit:

* https://github.com/smartsdk/ngsi-timeseries-api/blob/master/src/reporter/reporter.py#L172
* https://github.com/smartsdk/ngsi-timeseries-api/blob/master/src/translators/factory.py#L28
* https://github.com/smartsdk/ngsi-timeseries-api/blob/master/src/translators/crate.py#L1029
* https://github.com/smartsdk/ngsi-timeseries-api/blob/master/src/translators/base_translator.py#L19

Notice the use of a [context manager](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager)

If that's what you were worrying about I suggest we get rid of this sentence "but it also means that you need to redeploy ql in case of major crate update" since it's going to be confusing.

Reply:
then we can remove the comment, still is a query that consume time and 99% with the same result, so if we have somewhere this a static value that is not refreshed every notification, may be a better option.

Reply:
I updated the TODO text

Reply:
yep, excellent point, I opened a separate issue just for that: #302